rules/awsrules/aws_resource_missing_tags.go

```diff
@@ -320,6 +339,135 @@ func (r *AwsResourceMissingTagsRule) Check(runner *tflint.Runner) error {
 	return nil
 }
 
+// awsAutoscalingGroupTag is used by go-cty to evaluate tags in aws_autoscaling_group resources
+// The type does not need to be public, but its fields do
+// https://github.com/zclconf/go-cty/blob/master/docs/gocty.md#converting-to-and-from-structs
+type awsAutoscalingGroupTag struct {
+	Key               string `cty:"key"`
+	Value             string `cty:"value"`
+	PropagateAtLaunch bool   `cty:"propagate_at_launch"`
+}
+
+// checkAwsAutoScalingGroups handles the special case for tags on AutoScaling Groups
+// See: https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/autoscaling_tags.go
+func (r *AwsResourceMissingTagsRule) checkAwsAutoScalingGroups(runner *tflint.Runner, config awsResourceTagsRuleConfig) error {
+	resourceType := "aws_autoscaling_group"
+
+	for _, resource := range runner.LookupResourcesByType(resourceType) {
+		asgTagBlockTags, tagBlockLocation, err := r.checkAwsAutoScalingGroupsTag(runner, config, resource)
+		if err != nil {
+			return err
+		}
+
+		asgTagsAttributeTags, tagsAttributeLocation, err := r.checkAwsAutoScalingGroupsTags(runner, config, resource)
+		if err != nil {
+			return err
+		}
+
+		var location hcl.Range
+		tags := make(map[string]string)
+		switch {
+		case len(asgTagBlockTags) > 0 && len(asgTagsAttributeTags) > 0:
+			issue := fmt.Sprintf("Only tag block or tags attribute may be present, but found both")
+			runner.EmitIssue(r, issue, resource.DeclRange)
+			return nil
+		case len(asgTagBlockTags) == 0 && len(asgTagsAttributeTags) == 0:
+			r.emitIssue(runner, map[string]string{}, config, resource.DeclRange)
+			return nil
+		case len(asgTagBlockTags) > 0 && len(asgTagsAttributeTags) == 0:
+			tags = asgTagBlockTags
+			location = tagBlockLocation
+		case len(asgTagBlockTags) == 0 && len(asgTagsAttributeTags) > 0:
+			tags = asgTagsAttributeTags
+			location = tagsAttributeLocation
+		}
+
+		return runner.EnsureNoError(err, func() error {
+			r.emitIssue(runner, tags, config, location)
+			return nil
+		})
+	}
+	return nil
+}
+
+// checkAwsAutoScalingGroupsTag checks tag{} blocks on aws_autoscaling_group resources
+func (r *AwsResourceMissingTagsRule) checkAwsAutoScalingGroupsTag(runner *tflint.Runner, config awsResourceTagsRuleConfig, resource *configs.Resource) (map[string]string, hcl.Range, error) {
+	tags := make(map[string]string)
+	body, _, diags := resource.Config.PartialContent(&hcl.BodySchema{
+		Blocks: []hcl.BlockHeaderSchema{
+			{
+				Type: tagBlockName,
+			},
+		},
+	})
+	if diags.HasErrors() {
+		return tags, (hcl.Range{}), diags
+	}
+
+	for _, tagBlock := range body.Blocks {
+		attributes, diags := tagBlock.Body.JustAttributes()
+		if diags.HasErrors() {
+			return tags, tagBlock.DefRange, diags
+		}
+
+		var key string
+		err := runner.EvaluateExpr(attributes["key"].Expr, &key)
```

Comment:
It has a possibility of a null pointer exception.

Reply:
d583da6